### DISCUSSION TITLE
Can I use the steiger linter with lint-staged?

### DISCUSSION THREAD

**Discussion by OhSeungWan** [feature-sliced/steiger#97] [Q&A] [answered]
hi, I wanted to use the Steiger linter with lint-staged for pre-commit using husky.

So I tried it by adding the lint-staged option in package.json like this:

```
 "lint-staged": {
    "*.{js,ts,tsx,jsx,json}": [
      "prettier --write",
      "jest --ci --passWithNoTests",
      // "steiger ",
      // "steiger ./src",
      // "steiger src"
    ]
  },
```
----------
The error output to git log is as follows:

✖ steiger ./src:
steiger [options] <path>

Options:
  -w, --watch watch filesystem changes [boolean]
      --fix apply auto-fixes [boolean]
  -h, --help display help message [boolean]
  -v, --version Show version number [boolean]

Pass only one path to watch
Linting failed! Commit aborted.

-------------

Sorry if this is a stupid question, but I would still appreciate your help.😔

**Comment by illright** [comment, marked as answer]
Hello, thanks for the question! I don't think `lint-staged` is the right tool to run Steiger, simply because of its specifics. The goal of `lint-staged` is to let you check the changed files individually, while Steiger is built to run on the entire project, not only the changed files.

If you're using a pre-commit hook to validate your project before committing, then you should include the check alongside the call to `lint-staged`. For example, with Husky, the script could look like this:
```
npx lint-staged
npx steiger ./src
```

**Reply by OhSeungWan** [reply]
@illright Thank you for your reply.

I understand the purpose of steiger as you explained. 
I will try to apply it as you suggested.

thank you!
